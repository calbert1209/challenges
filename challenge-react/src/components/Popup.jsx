import React from 'react';
import styled from 'styled-components';
import Button from './Button';

const Overlay = styled.div`
  position: fixed;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  background: rgba(0, 0, 0, 0.7);
  transition: opacity 500ms;
  visibility: hidden;
  visibility: ${(props) => (props.visible ? 'visible' : 'hidden')};
  opacity: ${(props) => (props.visible ? 1 : 0)};
  z-index: 99;
`;

const PopupWrapper = styled.div`
  margin: 70px auto;
  padding: 20px;
  background: #fff;
  border-radius: 5px;
  width: 30%;
  position: relative;
  transition: all 5s ease-in-out;
  text-align: center;
`;

const Popup = (props) => {
  return (
    <Overlay visible={props.visible}>
      <PopupWrapper>
        <h2>{props.message}</h2>
        <Button id={"close-popup-btn"} onClick={props.onClose}>Close</Button>
      </PopupWrapper>
    </Overlay>
  );
};

export default Popup;
