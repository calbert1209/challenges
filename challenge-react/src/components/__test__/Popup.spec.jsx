import React from 'react';
import TestRenderer from 'react-test-renderer';
import Popup from '../Popup';
describe('Button', () => {
  describe('render correctly', () => {
    it('visible true', () => {
      const testRenderer = TestRenderer.create(
        <Popup visible={true} message={'test'} onClose={jest.fn()} />
      );

      expect(testRenderer.toJSON()).toMatchSnapshot();
    });
    it('visible false', () => {
      const testRenderer = TestRenderer.create(
        <Popup visible={false} message={'test'} onClose={jest.fn()} />
      );

      expect(testRenderer.toJSON()).toMatchSnapshot();
    });
  });
  it('close popup', () => {
    const mockOnClose = jest.fn();
    const testRenderer = TestRenderer.create(
      <Popup visible={true} message={'test'} onClose={mockOnClose} />
    );
    const testInstance = testRenderer.root;
    testInstance.findByProps({ id: 'close-popup-btn' }).props.onClick();
    expect(mockOnClose).toHaveBeenCalled();
  });
});
