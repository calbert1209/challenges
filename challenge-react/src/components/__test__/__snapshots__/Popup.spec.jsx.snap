// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button render correctly visible false 1`] = `
<div
  className="sc-gsTCUz gnBEhQ"
>
  <div
    className="sc-dlfnbm LtJEX"
  >
    <h2>
      test
    </h2>
    <button
      className="sc-bdfBwQ eCLJup"
      id="close-popup-btn"
      onClick={[MockFunction]}
    >
      Close
    </button>
  </div>
</div>
`;

exports[`Button render correctly visible true 1`] = `
<div
  className="sc-gsTCUz jXWlsB"
>
  <div
    className="sc-dlfnbm LtJEX"
  >
    <h2>
      test
    </h2>
    <button
      className="sc-bdfBwQ eCLJup"
      id="close-popup-btn"
      onClick={[MockFunction]}
    >
      Close
    </button>
  </div>
</div>
`;
